Administrator Guide/Split brain and ways to deal with it.md

```diff
@@ -0,0 +1,61 @@
+3. B2 comes up and B1 goes down. Quorum is met. But when a write request comes, AFR sees that B2 & B3's pending xattrs blame each other and therefore the write is not allowed and is failed with EIO.
+
+Command to create a replica 3 volume:
+```sh
+$gluster volume create <volname> replica 3 host1:brick1 host2:brick2 host3:brick3
+```
+There is a corner case even with replica 3 volumes where the file can end up in a split-brain. AFR usually takes range locks for the {offset, length} of the write. If 3 writes happen on the same file at non-overlapping {offset, length} and each write fails on (only) one different brick, then we have AFR xattrs of the file blaming each other.
+
+### 2. Arbiter volume:
+This is also an replica 3 volume where the third brick of the replica is automatically configured as an arbiter node. This means that the third brick stores only the file name and metadata, but not any data. This will help in avoiding split brain while providing the same level of consistancy as a normal replica 3 volume.
+
+Command to create a arbiter volume:
+```sh
+$gluster volume create <volname> replica 3 arbiter 1 host1:brick1 host2:brick2 host3:brick3
+```
+
+The only difference in the command is, we need to add one more keyword ``` arbiter 1 ``` after the replica count. Since it is also a replica 3 volume, the cluster.quorum-type option is set to auto by default and atleast 2 bricks should be up to satisfy the quorum and allow writes.
+Since the arbiter brick has only name and metadata of the files, there are some more checks to guarrantee consistancy. Arbiter works as follows:
+
+1. Clients take full file locks while writing (replica 3 takes range locks).
+2. If 2 bricks are up and if one of them is the arbiter, and it blames the other up brick, then all FOPs will fail with ENOTCONN (Transport endpoint is not connected). If the arbiter doesn't blame the other brick, FOPs will be allowed to proceed.
+3. If 2 bricks are up and the arbiter is down, then FOPs will be allowed.
+4. If only one brick is up, then client-quorum is not met and the volume becomes EROFS.
+5. In all cases, if there is only one source before the FOP is initiated and if the FOP fails on that source, the application will receive ENOTCONN.
+
+You can find more details on arbiter [here](https://gluster.readthedocs.io/en/latest/Administrator%20Guide/arbiter-volumes-and-quorum/).
+
+### Differences between replica 3 and arbiter volumes:
+1. In case of a replica 3 volume, we store the entire file in all the bricks and it is recommended to have bricks of same size. But in case of arbiter, since we do not store data, the size of the arbiter brick is comparitively lesser than the other bricks.
+2. Arbiter is a state between replica 2 and replica 3 volume. If we have only arbiter and one of the other brick is up and the arbiter brick blames the other brick, then we can not proceed with the FOPs.
+3. Replica 3 gives high availability compared to arbiter.
```

Comment:
..because unlike in arbiter, replica 3 has a full copy of the data in all 3 bricks.